### PR TITLE
Upgrades Blacklight, Rails, and Ruby.

### PR DESCRIPTION
**Pull request by bwatson78**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -18,7 +19,7 @@ jobs:
                 default: '4'
             mysql_version:
                 type: string
-                default: '5.7.22'
+                default: '8.0.33'
```
Comment (abelemlih) [inline .circleci/config.yml:22]: Are we upgrading MySQL on the deployed instances as well?

Comment (bwatson78) [reply .circleci/config.yml:22]: Yep, good catch. Thanks!

File: config/environments/production.rb [hunk 2]
```diff
@@ -15,52 +17,52 @@
   config.consider_all_requests_local       = false
   config.action_controller.perform_caching = true
 
-  # Attempt to read encrypted secrets from `config/secrets.yml.enc`.
-  # Requires an encryption key in `ENV["RAILS_MASTER_KEY"]` or
-  # `config/secrets.yml.key`.
-  config.read_encrypted_secrets = true
```
Comment (abelemlih) [inline config/environments/production.rb:21]: What does this change do exactly?

Comment (bwatson78) [reply config/environments/production.rb:21]: This came in with the `Adds all new/adjusted files.` commit that was focused on matching all of the code changes that the newer Blacklight version implemented when changing Rails versions. In this case, I made sure that all of the matched code wasn't altered by us beforehand.

File: config/initializers/assets.rb [hunk 3]
```diff
@@ -2,12 +2,10 @@
 # Be sure to restart your server when you modify this file.
 
 # Version of your assets, change this if you want to expire all your assets.
-Rails.application.config.assets.version = '1.0'
+Rails.application.config.assets.version = "1.0"
 
 # Add additional assets to the asset load path.
 # Rails.application.config.assets.paths << Emoji.images_path
-# Add Yarn node_modules folder to the asset load path.
-Rails.application.config.assets.paths << Rails.root.join('node_modules')
```
[the remark this answers is not in the document]
Comment (bwatson78) [reply config/initializers/assets.rb:10]: This is in the same commit (`Adds all new/adjusted files.`) I'm guessing that, with the new Rails version, `node_modules` is either loaded by default or not needed. But this matches the new Blacklight version's behavior.

**Review by abelemlih — APPROVED** [review]

**Pull request merged**